### ISSUE TITLE
Bug: (fill in) ESLint couldn't find the plugin "eslint-plugin-prettier".

### ISSUE THREAD

**Issue by da-du** [eslint/eslint#16694]
### Environment

Node version: v18.2.0
npm version: v8.9.0
Local ESLint version: Not found
Global ESLint version: Not found
Operating System: darwin 21.6.0

### What parser are you using?

Default (Espree)

### What did you do?

My project depends on "@qunhe/eslint-config-saas", "@qunhe/eslint-config-saas" depends on "eslint-plugin-prettier", when I run "eslint", an error is reported

<img width="1169" alt="image" src="https://user-images.githubusercontent.com/106589025/209301940-3ba8ed68-3a67-426f-bce9-bd29d762a280.png">

My node_modules is like this：
```
-- node_modules
  -- .bin
  -- @qunhe
    -- eslint-config-saas
      -- node_modules
        -- .bin
        -- eslint-plugin-prettier
```

### What did you expect to happen?

I don't want to depend on "eslint-plugin-prettier" in my project, because I don't need it

### What actually happened?

yarn run v1.22.19
warning ../../package.json: No license field
$ eslint --ext js,jsx,ts,tsx src/ --fix

Oops! Something went wrong! :(

ESLint: 8.30.0

ESLint couldn't find the plugin "eslint-plugin-prettier".

(The package "eslint-plugin-prettier" was not found when loaded as a Node module from the directory "/Users/XXX/project/saas-statistics".)

It's likely that the plugin isn't installed correctly. Try reinstalling by running the following:

    npm install eslint-plugin-prettier@latest --save-dev

The plugin "eslint-plugin-prettier" was referenced from the config file in ".eslintrc.js » @qunhe/eslint-config-saas".

If you still can't figure out the problem, please stop by https://eslint.org/chat/help to chat with the team.

error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by aladdin-add** [comment]
with the current(eslintrc) config system, the plugin should be a peerdep of the shared config, not a normal dep.

a workaround is to run `npx yarn-deduplicate yarn.lock`.

**Comment by mdjermanovic** [comment]
> I don't want to depend on "eslint-plugin-prettier" in my project, because I don't need it

With the current config system, you have to specify all eslint plugins as dev-dependencies in your project. This will not be necessary with the [new config system](https://eslint.org/docs/latest/user-guide/configuring/configuration-files-new).
